src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -209,6 +209,27 @@ public static void RegisterMSBuildPath(string[] msbuildSearchPaths)
             // AssemblyResolve event can fire multiple times for the same assembly, so keep track of what's already been loaded.
             var loadedAssemblies = new Dictionary<string, Assembly>();
 
+            // MSBuild can be loaded from the x86 or x64 folder. Before 17.0, it looked next to the executing assembly in some cases and constructed a path that assumed x86 in others.
+            // This overrides the latter assumption to let it find the right MSBuild.
+            foreach (string path in msbuildSearchPaths)
```

Comment:
Do we not need to do the `amd64` folder removal dance here too, in case someone passed in a 64-bit folder?

Can we do this only for <17.0?

Reply:
Isn't that handled in MSBuild? I'm looking at TryFromMSBuildExeUnderVisualStudio, and it calculates visualStudioRoot by checking whether there's an amd64 and going one extra level up if so.

On the other path, it assumes a standalone MSBuild.exe, so I wasn't sure what assumptions we could actually make about it.

Reply:
Doing it for < 17.0 sounds reasonable. Little more expensive perf-wise, but that isn't a big deal in Locator.

Reply:
Are you looking at MSBuild 16 or 17 source? That was a thing we fixed, right?